chapter_linear-networks/softmax-regression-scratch.md

```diff
@@ -432,7 +432,7 @@ def train_epoch_ch3(net, train_iter, loss, updater):  #@save
 
 ```{.python .input}
 #@tab tensorflow
-def train_epoch_ch3(net, train_iter, loss, updater):  #@save
+def train_epoch_ch3(net, train_iter, loss, updater, params=None):  #@save
```

Comment:
why we need `params`? can we remove it to make the api consistent.

Reply:
In dropout section, net is a custom function which doesn’t have trainable_variables and its optimizer/updater is from tf.keras instead of using the Updater class you created in previous section. We need to pass params since we cannot attach it to either the optimizer or the custom model function.

Reply:
Can you use the `Updater` class instead?

Reply:
Okay I switched to use `Updater` here. I originally thought that since the PyTorch implementation here uses `torch.optim.SGD` so it might be better to support `tf.keras.Optimizer` in `train_ch3()`.